### PR TITLE
BAH-2464 - Refactoring code from Bahmni-Core [WIP]

### PR DESCRIPTION
**Pull request by sivareddyp**
JIRA - [https://bahmni.atlassian.net/browse/BAH-2464](https://bahmni.atlassian.net/browse/BAH-2464)

- Upgraded OpenMRS version to 2.5.7 and related dependencies
- Removed deprecated code from Bahmni-Core
- Moved duplicate code from Bahmni-Core
- Moved code pertaining to `openmrs/ws/rest/v1/bahmnicore/search/patient` in Bahmni-Core to `openmrs/ws/rest/v1/bahmni/search/patient` in bahmni-commons so that Bahmni-Commons is the only repo contains code related to rest end point /search/patient [The same end point is used in appointments]
- Moved code pertaining to `openmrs/ws/rest/v1/bahmnicore/search/patien/lucene` in Bahmni-Core to `openmrs/ws/rest/v1/bahmni/search/patient/lucene` in bahmni-commons so that Bahmni-Commons is the only repo contains code related to rest end point /search/patient/lucene [The same end point is used in appointments]

### CODE REVIEW

**Comment by binduak** [comment]
@sivareddytw few pointers from the code review
1. Can we also move Lucene search code to bahmnicommoms from bahmni-core
2. `bahmniVisitLocationService` has duplicate beans. Can we create a tech debt card to handle this
3. test coverage in the bahmnicommons needs to be improved. This can also added to the above card.

**Comment by sivareddyp** [comment]
> * Can we also move Lucene search code to bahmnicommoms from bahmni-core

yes @binduak. Lucene search moved to bahmni-commons. Please check these commits 
[bahmni-commons commit](https://github.com/Bahmni/openmrs-module-bahmnicommons/pull/13/commits/ea91fcc2b577bbde36cc4f5a28c60d14c669f5e9). 
[bahmni-core commit](https://github.com/Bahmni/bahmni-core/pull/164/commits/f233895a84e4603d28c4bf859a643a446e1cabf5)

**Comment by sivareddyp** [comment]
> * `bahmniVisitLocationService` has duplicate beans. Can we create a tech debt card to handle this

removed duplicate beans.

**Comment by sivareddyp** [comment]
> 3\. test coverage in the bahmnicommons needs to be improved. This can also added to the above card.

Card [BAH-2508](https://bahmni.atlassian.net/browse/BAH-2508) created for test coverage.

File: pom.xml [hunk 1]
```diff
@@ -69,20 +69,6 @@
 
     <dependencyManagement>
         <dependencies>
-            <!--Child modules-->
-            <dependency>
-                <groupId>org.bahmni.module</groupId>
```
Comment (binduak) [inline pom.xml:74]: @sivareddytw why the api and omod dependency removed from main pom.xml file ?

Comment (sivareddyp) [reply pom.xml:74]: @binduak bahmni-commons-omod is removed as it not used as a dependency in any of modules. 
bahmni-commons-api is retained

**Review by binduak — APPROVED** [review]

**Comment by sivareddyp** [comment]
Closing this PR.
Raising PRs for OpenMRS upgrade, patient search, lucene search and EMail Changes

**Pull request closed without merging**